options/commonOptions.go

```diff
@@ -0,0 +1,593 @@
+package options
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"time"
+
+	dtlsServer "github.com/plgd-dev/go-coap/v2/dtls/server"
+	"github.com/plgd-dev/go-coap/v2/message/pool"
+	"github.com/plgd-dev/go-coap/v2/mux"
+	"github.com/plgd-dev/go-coap/v2/net/blockwise"
+	"github.com/plgd-dev/go-coap/v2/net/client"
+	"github.com/plgd-dev/go-coap/v2/net/monitor/inactivity"
+	"github.com/plgd-dev/go-coap/v2/pkg/runner/periodic"
+	tcpClient "github.com/plgd-dev/go-coap/v2/tcp/client"
+	tcpServer "github.com/plgd-dev/go-coap/v2/tcp/server"
+	udpClient "github.com/plgd-dev/go-coap/v2/udp/client"
+	udpServer "github.com/plgd-dev/go-coap/v2/udp/server"
+)
+
+type Handler interface {
+	tcpClient.HandlerFunc | udpClient.HandlerFunc
+}
+
+// HandlerFuncOpt handler function option.
+type HandlerFuncOpt[H Handler] struct {
+	h H
+}
+
+func panicForInvalidHandlerFunc(t, exp any) {
+	panic(fmt.Errorf("invalid HandlerFunc type %T, expected %T", t, exp))
+}
+
+func (o HandlerFuncOpt[H]) TCPServerApply(cfg *tcpServer.Config) {
+	switch v := any(o.h).(type) {
+	case tcpClient.HandlerFunc:
+		cfg.Handler = v
```

Comment:
We could create a base config struct with all the shared members by `tcpServer.Config`, `tcpClient.Config`, `udpServer.Config`, `dtlsServer.Config` and `udpClient.Config`. It seems to be something like this:

```go
type CommonConfig struct {
	Ctx                            context.Context
	Handler                        HandlerFunc
	Errors                         ErrorFunc
	GoPool                         GoPoolFunc
	PeriodicRunner                 periodic.Func
	MessagePool                    *pool.Pool
	GetToken                       client.GetTokenFunc
	CreateInactivityMonitor        func() inactivity.Monitor
	MaxMessageSize                 uint32
	BlockwiseTransferTimeout       time.Duration
	BlockwiseSZX                   blockwise.SZX	
	BlockwiseEnable                bool
}
```

Reply:
Fixed